util/reggen/countermeasure.py

```diff
@@ -0,0 +1,71 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Dict, Sequence
+
+from .lib import check_keys, check_str, check_list
+
+# TO BE DEFINED
+CM_ASSETS = [
+    'FOO'
+]
+
+# TO BE DEFINED
+CM_TYPES = [
+    'BAR'
+]
+
+
+class CounterMeasure:
+    def __init__(self, name: str, desc: str):
+        self.name = name
+        self.desc = desc
+
+    @staticmethod
+    def from_raw(what: str, raw: object, bname: str) -> 'CounterMeasure':
+        rd = check_keys(raw, what, ['name', 'desc'], [])
+
+        name = check_str(rd['name'], 'name field of ' + what)
+        desc = check_str(rd['desc'], 'desc field of ' + what)
+
+        name_split = name.split('.')
```

Comment:
Recommend changing this to:
```python 
    try:
          self.block, self.asset, self.type = name.split('.')
    except ValueError:
          raise ValueError(...)
```
This not only checks the split yielded a list of 3, but also creates a named lookup for the asset and type. +1 to @rswarbrick's suggestion to maintain these 3 fields individually.

Reply:
Good idea - changed!